client/components/sidebar.vue

```diff
@@ -250,21 +252,26 @@
       },
 
       createComment(highlightKey) {
-        if (highlightKey === this.commentToAdd.highlightKey) {
-          Comment.create(this.commentToAdd);
-          this.commentToAdd = null;
+        const commentToAdd = commentsToAdd.find((x) => {
+          return x.highlightKey === highlightKey;
+        });
+        if (commentToAdd) {
+          Comment.create(commentToAdd);
```

Comment:
Maybe first filter and then call a method.

Also, what happens in JavaScript if you modify the list over which you are iterating while iterating?

Reply:
Done.